daemon/graphdriver/copy/copy.go

```diff
@@ -152,8 +152,10 @@ func DirCopy(srcDir, dstDir string, copyMode Mode, copyXattrs bool) error {
 
 		isHardlink := false
 
-		switch f.Mode() & os.ModeType {
-		case 0: // Regular file
+		fMode := f.Mode()
+
+		switch {
```

Comment:
perhaps
```go
switch mode := f.Mode(); {
      case mode.IsRegular():
              ....
      case mode.IsDir():
              ....
      case mode&os.ModeSymlink != 0:
              ....
      ....
}
```
The differences are, this code
* uses standard `os.IsXxx()` checkers where possible;
* does not introduce a variable seen outside of `switch{}`.

Reply:
Ah, yes, makes sense: updated 👍